### ISSUE TITLE
Cannot get coverage report if c8 is running in a sub-directory & src is one level up.

### ISSUE THREAD

**Issue by timganter** [bcoe/c8#527]
* **9.1.0**:
* **MacOS**:

Hello! When I have c8 running from a sub-directory, I can't seem to get it to report coverage for src code that is one level up from where c8 is running. 

- I have created this repo that reproduces the issue. https://github.com/timganter/c8-sub-directory
- Am I just configuring c8 incorrectly 🤞 ? [Here's my config](https://github.com/timganter/c8-sub-directory/blob/main/sub-directory/.c8rc.json). 
- Or is this a bug? 
- Here is a video showing the issue that can be reproduced [using this repo](https://github.com/timganter/c8-sub-directory). 

https://github.com/bcoe/c8/assets/1911000/e871d7cd-9e92-492e-b178-a81ffe2d4b18

**Comment by ericmorand** [comment]
I have the same issue: https://github.com/bcoe/c8/issues/498

**Comment by cenfun** [comment]
For `sub-directory`, the source file `index.js` is outside of the current working directory, so it was excluded. see https://github.com/istanbuljs/test-exclude/blob/master/index.js#L90-L92

**Comment by ericmorand** [comment]
Yes. Actually, c8 doesn't consider the location of the configuration file as the base path to resolve the sources to cover. This makes declaring the sources there useless except when executing c8 from the same directory as the one where the configuration file is located.

There are some other issues too that prevent c8 from working properly outside of the "root" directory: include and exclude are also not relative to the configuration file location, and they basically contradict the `src` configuration and the source map of the executed module. Basically c8 is plagued by the same design decisions as `nyc` but to an even greater extend.

**Comment by cenfun** [comment]
@ericmorand You can try [mcr CLI](https://github.com/cenfun/monocart-coverage-reports?#command-line)
```
// mcr.config.js
export default {
    name: 'My Coverage Report',
    reports: [
        'v8',
        'console-details'
    ],
    outputDir: './coverage-reports',
    entryFilter: {
        '**/node_modules/**': false,
        '**/src/**': true
    },
    sourceFilter: {
        '**/src/**': true
    }
};

```
```
mcr npx mocha
┌────────────┬──────────┬────────────┬──────────┬───────────┬──────────┬─────────────────┐
│ Name       │    Bytes │ Statements │ Branches │ Functions │    Lines │ Uncovered Lines │
├────────────┼──────────┼────────────┼──────────┼───────────┼──────────┼─────────────────┤
│ src        │          │            │          │           │          │                 │
│ └ index.js │ 100.00 % │   100.00 % │          │  100.00 % │ 100.00 % │                 │
├────────────┼──────────┼────────────┼──────────┼───────────┼──────────┼─────────────────┤
│ Summary    │ 100.00 % │   100.00 % │          │  100.00 % │ 100.00 % │                 │
└────────────┴──────────┴────────────┴──────────┴───────────┴──────────┴─────────────────┘
cd sub-directory
mcr npx mocha -c ../mcr.config.js
┌────────────┬──────────┬────────────┬──────────┬───────────┬──────────┬─────────────────┐
│ Name       │    Bytes │ Statements │ Branches │ Functions │    Lines │ Uncovered Lines │
├────────────┼──────────┼────────────┼──────────┼───────────┼──────────┼─────────────────┤
│ src        │          │            │          │           │          │                 │
│ └ index.js │ 100.00 % │   100.00 % │          │  100.00 % │ 100.00 % │                 │
├────────────┼──────────┼────────────┼──────────┼───────────┼──────────┼─────────────────┤
│ Summary    │ 100.00 % │   100.00 % │          │  100.00 % │ 100.00 % │                 │
└────────────┴──────────┴────────────┴──────────┴───────────┴──────────┴─────────────────┘
```

**Comment by skullpsg** [comment]
@timganter Can you try with  --allowExternal on the cmd. It would work.

**Comment by ericmorand** [comment]
@cenfun thank you..I think you got it right by having both a source and an entry filter approach. Can you elaborate a bit on the difference between the two?

**Comment by cenfun** [comment]
@ericmorand Both can generate coverage reports from raw v8 coverage data.
`c8` uses the converter `v8-to-istanbul`, and it can only generate Istanbul reports.
`mcr` uses its own converter (can analyze AST). In addition to generating Istanbul reports, it can also generate V8-style reports.

**Comment by ericmorand** [comment]
I opened an issue in your repository to move the conversation there: https://github.com/cenfun/monocart-coverage-reports/issues/22

**Comment by timganter** [comment]
@skullpsg sorry, no 🎲 . You can try it out on my repo here: https://github.com/timganter/c8-sub-directory?tab=readme-ov-file#steps-to-reproduce


<img width="507" alt="allowExternal" src="https://github.com/bcoe/c8/assets/1911000/10a1051b-40d8-4b34-ae5b-e4b2adad090d">
